### PR TITLE
Add projects argument under browse-osp

### PR DESCRIPTION
**Pull request by adrianfusco**
Used urllib to extract easily the project names in every URL provided by the  section of the packages.
In this way we can parse the Git URL (extracted with the net_loc) and receive the projecs (Which is the path) following the [URL Syntactic Components](https://www.rfc-editor.org/rfc/rfc1808.html#section-2.1).

### CODE REVIEW

File: znoyder/browser.py [hunk 1]
```diff
@@ -104,6 +122,10 @@ def process_arguments(argv=None) -> Namespace:
     packages.add_argument('--tag', dest='tag')
     packages.add_argument('--upstream', dest='upstream')
 
+    projects = subparsers.add_parser('projects', help='', parents=[common])
+    projects.add_argument('--git_url', dest='git_url', default=False,
```
Comment (bregman-arie) [inline znoyder/browser.py:126]: can you change the argument from --git_url to simply --git ?

**Review by bregman-arie — CHANGES_REQUESTED** [review]

**Comment by sdatko** [comment]
Hmm, after a few thoughts, I have a doubt it should be done in this place.
The ospinfo itself does not provide `projects` *per se*, so why to add it, with another custom set of commands, while what we do is we process the packages information and so we can simply take basenames of `znoyder browse-osp packages --output osp-patches`?
Can we discuss tomorrow again what's the goal and how we want to use this?

I would vote more for adding filter on osp-patches to get_packages() if we really need it and not introduce additional command like that here.

**Comment by adrianfusco** [comment]
> Hmm, after a few thoughts, I have a doubt it should be done in this place. The ospinfo itself does not provide `projects` _per se_, so why to add it, with another custom set of commands, while what we do is we process the packages information and so we can simply take basenames of `znoyder browse-osp packages --output osp-patches`? Can we discuss tomorrow again what's the goal and how we want to use this?
> 
> I would vote more for adding filter on osp-patches to get_packages() if we really need it and not introduce additional command like that here.

Yes, let discuss it tomorrow.

File: README.md [hunk 2]
```diff
@@ -35,6 +35,12 @@ To list URLs to repositories of packages that belong to network component:
 znoyder browse-osp packages --component network --tag osp-17.0 --output osp-patches
 ```
 
+There is a argument used in the output to filter project name:
```
Comment (sdatko) [inline README.md:38]: Maybe better: There is an additional key `osp-project` introduced that contains downstream repository name
?

**Review by sdatko — APPROVED** [review]

File: README.md [hunk 3]
```diff
@@ -35,6 +35,12 @@ To list URLs to repositories of packages that belong to network component:
 znoyder browse-osp packages --component network --tag osp-17.0 --output osp-patches
 ```
 
+There is a argument used to filter name projects. It will extract the information of the packages and get all the project names. We can filter in base of Git URL.
```
[the remark this answers is not in the document]
Comment (sdatko) [reply README.md]: Och, that was a comment to previous version – I did not submit it properly before, sorry.

File: znoyder/browser.py [hunk 4]
```diff
@@ -81,6 +82,23 @@ def get_releases(**kwargs):
     return releases
 
 
+def get_projects(**kwawrgs) -> list:
+    packages = get_packages()
+    urls_components = [package['osp-patches'] for package in packages
+                       if package['osp-patches'] != '']
```
[the remark this answers is not in the document]
Comment (sdatko) [reply znoyder/browser.py]: Och, that was a comment to previous version – I did not submit it properly before, sorry.

File: README.md (hunk 2 above)
Comment (adrianfusco) [reply README.md]: You're right. It's a better explanation.

Change done.

**Pull request merged**